src/components/pages/MainQuizPage/EndQuizButton/EndQuizButton.tsx

```diff
@@ -9,22 +8,25 @@ const EndQuizButton = () => {
 
     const [showModal, setShowModal] = useState(false);
     const navigate = useNavigate();
-
-    const onShowModalHandler = () => setShowModal(!showModal);
+    const onCloseModalHandler = () => setShowModal(!showModal);
     const onNavigateToResultHandler = () => navigate(START);
 
     return (
         <>
-            <StyledButton onClickHandler={onShowModalHandler} text="Finish test" />
-            {
-                showModal && createPortal(
-                    <ModalPage
-                        setShowModal={onShowModalHandler}
-                        navigateToResult={onNavigateToResultHandler}
-                    />,
-                    document.body
-                )
-            }
+            <StyledButton onClickHandler={onCloseModalHandler} text="Finish test" />
+            <ModalPage
+                isOpen={showModal}
+                setShowModal={onCloseModalHandler}
+                navigateToResult={onNavigateToResultHandler}
```

Comment:
could you please rename `navigateToResult` to `onConfirm` since you use this on the confirm button